modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/src/Runtime/LegacyCliOptions.cs

```diff
@@ -244,7 +242,15 @@ public class LegacyCliOptions : Dictionary<string, string>, IAgentConfigProvider
         /// <summary>
         /// The model of the CLI arguments.
         /// </summary>
-        public LegacyCliModel LegacyCliModel { get; }
+        public LegacyCliModel LegacyCliModel {
+            get {
+                if (_legacyCliModel == null) {
+                    _legacyCliModel = ToLegacyCliModel();
+                }
```

Comment:
Simplify like this?
_legacyCliModel ??= ToLegacyCliModel();

Reply:
I would say it's preference. ;-)

Reply:
Agree ;)